### ISSUE TITLE
Origin distance variation. 144 origin points.

### ISSUE THREAD

**Issue by jojkaart** [cloudflare/cloudflare-blog#32]
Edit: This is in reply to https://blog.cloudflare.com/computing-euclidean-distance-on-144-dimensions/
I believe the origin distance method, if used with up to 144 *different* origin points, could be used to find whether there is a match (and the match) with only 144 distance calculations in the worst case. Probably much less on average, though. Also, when there's no match, I expect that'd be apparent after just a few comparisons in most cases.

Basically, you'd be doing something very similar to triangulation for each search. Just in 144D instead of 2D or 3D. https://en.wikipedia.org/wiki/Triangulation_(surveying)

**Comment by majek** [comment]
Then the complexity is in finding the intersection of the 144 calculations. So:
 - first search from origin returned say 80% of all database (800k items)
 - second search returned 80% of all database (800k items)
And so on
So, you end up with 144 sets of point names (say uint32), then you need to find the intersection of these unordered (they are ordered by the distance form origin, so _not_ ordered by name) sets. Finding intersection becomes non trivial.
